### PR TITLE
Update file_selector_* to the latest pigeon

### PR DESCRIPTION
**Pull request by natebosch**
The current version is incompatible with the latest analyzer, which is a blocker for getting the latest mockito, which is a blocker for getting the latest test_api, which is pinned by flutter_test and will be breaking in the next package roll.

Also update to the latest `build_runner` and use a range constraint since the older `build_runner` is also incompatible with the latest `analyzer`.

### CODE REVIEW

File: packages/file_selector/file_selector_ios/ios/Classes/messages.g.h [hunk 1]
```diff
@@ -17,21 +19,18 @@ NS_ASSUME_NONNULL_BEGIN
 /// `init` unavailable to enforce nonnull fields, see the `make` class method.
 - (instancetype)init NS_UNAVAILABLE;
 + (instancetype)makeWithUtis:(NSArray<NSString *> *)utis
-         allowMultiSelection:(NSNumber *)allowMultiSelection;
-@property(nonatomic, strong) NSArray<NSString *> *utis;
-@property(nonatomic, strong) NSNumber *allowMultiSelection;
+    allowMultiSelection:(NSNumber *)allowMultiSelection;
+@property(nonatomic, strong) NSArray<NSString *> * utis;
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_ios/ios/Classes/messages.g.h:23]: You need to format the PR: `dart run script/tool/bin/flutter_plugin_tools.dart format`

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/file_selector/file_selector_ios/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## NEXT
```
[the remark this answers is not in the document]
Comment (natebosch) [reply packages/file_selector/file_selector_ios/CHANGELOG.md:1]: I must have misread the wiki.

File: packages/file_selector/file_selector_ios/ios/Classes/messages.g.h (hunk 1 above)
Comment (natebosch) [reply packages/file_selector/file_selector_ios/ios/Classes/messages.g.h]: done

**Comment by stuartmorgan** [comment]
I pushed an update for the breaking changes affecting macOS.

**Comment by Hixie** [comment]
test-exempt: dependency roll

**Comment by stuartmorgan** [comment]
Oops, native tests. Fix incoming.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**